hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/BlockStreamBuilder.java

```diff
@@ -840,7 +844,10 @@ public BlockStreamBuilder contractID(@Nullable final ContractID contractID) {
     @NonNull
     @Override
     public BlockStreamBuilder exchangeRate(@Nullable final ExchangeRateSet exchangeRate) {
-        transactionResultBuilder.exchangeRate(exchangeRate);
+        // Block Stream doesn't include exchange rate in output (it's in state
+        // changes when it is updated), so store exchange rate for the
+        // translation context
```

Comment:
😳 How have our block <-> record comparisons been passing if the exchange rate hasn't been included?

Reply:
The exchange rate _was_ included in the block stream, but is removed in the latest version (as presented in this PR), as it's unnecessary extra data once block nodes exist (it can be provided from state changes much more efficiently).